operator/src/main/java/org/bf2/operator/managers/CapacityManager.java

```diff
@@ -203,8 +203,11 @@ private boolean releaseResources(Map<String, String> resourceData, String entryK
         return false;
     }
 
-    public synchronized Optional<OperandReadiness> claimResources(ManagedKafka managedKafka, String profile,
+    synchronized Optional<OperandReadiness> claimResources(ManagedKafka managedKafka, String profile,
             ManagedKafkaAgent agent) {
+        if (managedKafka.isReserveDeployment()) {
```

Comment:
Do we've to do the same for https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/blob/08a05a131c95d401bad591bcc2790c598d404221/operator/src/main/java/org/bf2/operator/managers/CapacityManager.java#L177 when a reserved instance is deleted (removed from the list of MKs)?

Reply:
By preventing the claim in theory a reserved instance should never make it into the list.  Always deleting covers a very unlikely corner case - that someone toggled the delete flag and change the instance from real to reserved at the same time.

Reply:
Right, thanks. I tested this and everything is working as expected for reserved instance.
I'll move along testing real deployments cases.